Indicators/Vortex.cs

```diff
@@ -12,121 +13,133 @@
-        decimal negativeVortex = sumVMMinus / sumTR;
-
-        PositiveVortexIndicator.Update(input.Time, positiveVortex);
-        NegativeVortexIndicator.Update(input.Time, negativeVortex);
-
-        // Depending on your test setup, you might want to return different values:
-        return positiveVortex; // If the test expects Positive Vortex
-        // return negativeVortex; // If the test expects Negative Vortex
-    }
-
-    /// <summary>
-    /// Gets the positive vortex indicator.
-    /// </summary>
-    public IndicatorBase<IndicatorDataPoint> PositiveVortexIndicator { get; } = new Identity("PositiveVortex");
-
-    /// <summary>
-    /// Gets the negative vortex indicator.
-    /// </summary>
-    public IndicatorBase<IndicatorDataPoint> NegativeVortexIndicator { get; } = new Identity("NegativeVortex");
+            PlusVI = new FunctionalIndicator<IndicatorDataPoint>("PlusVI",
+                input => CalculateVortexIndex(_plusVMSum, _atr.Current.Value * _period),
+                isReady => _plusVM.Count >= _period && _atr.IsReady,
+                () => { _plusVM.Clear(); _plusVMSum = 0m; });
 
-    /// <summary>
-    /// Gets a value indicating whether this indicator is ready and fully initialized.
-    /// </summary>
-    public override bool IsReady => _window.Count == _period + 1;
+            MinusVI = new FunctionalIndicator<IndicatorDataPoint>("MinusVI",
+                input => CalculateVortexIndex(_minusVMSum, _atr.Current.Value * _period),
+                isReady => _minusVM.Count >= _period && _atr.IsReady,
+                () => { _minusVM.Clear(); _minusVMSum = 0m; });
+        }
 
-    /// <summary>
-    /// Adds two VortexIndicator instances together.
-    /// </summary>
-    /// <param name="left">The left VortexIndicator.</param>
-    /// <param name="right">The right VortexIndicator.</param>
-    /// <returns>A new VortexIndicator instance representing the sum of the two indicators.</returns>
-    public static VortexIndicator operator +(VortexIndicator left, VortexIndicator right)
-    {
-        return new VortexIndicator(left._period + right._period);
-    }
+        protected override decimal ComputeNextValue(IBaseDataBar input)
+        {
+            if (_previousInput != null)
+            {
+                var plusVMValue = Math.Abs(input.High - _previousInput.Low);
+                var minusVMValue = Math.Abs(input.Low - _previousInput.High);
+
+                if (_plusVM.Count == _period)
+                {
+                    _plusVMSum -= _plusVM[0];
+                    _plusVM.RemoveAt(0);
+                }
+                if (_minusVM.Count == _period)
+                {
+                    _minusVMSum -= _minusVM[0];
+                    _minusVM.RemoveAt(0);
+                }
+
+                _plusVM.Add(plusVMValue);
+                _minusVM.Add(minusVMValue);
+
+                _plusVMSum += plusVMValue;
+                _minusVMSum += minusVMValue;
+            }
```

Comment:
This looks like they should be `WindowIndicator` or even better `Sum` indicators, which take the period as a parameter.